tests/study_tests/test_dataframe.py

```diff
@@ -190,19 +201,30 @@ def f(trial: Trial) -> Tuple[float, float]:
     ],
 )
 @pytest.mark.parametrize("multi_index", [True, False])
+@pytest.mark.parametrize("metric_names", [None, ["v0", "v1"]])
 def test_trials_dataframe_with_multi_objective_optimization_with_fail_and_pruned(
-    attrs: Tuple[str, ...], multi_index: bool
+    attrs: tuple[str, ...], multi_index: bool, metric_names: list[str] | None
 ) -> None:
     study = create_study(directions=["minimize", "maximize"])
+    if metric_names is not None:
+        study.set_metric_names(metric_names)
     study.add_trial(create_trial(state=TrialState.FAIL))
     study.add_trial(create_trial(state=TrialState.PRUNED))
     df = study.trials_dataframe(attrs=attrs, multi_index=multi_index)
 
     if multi_index:
         for i in range(2):
-            assert df.get("values")[0][i] is None
-            assert df.get("values")[1][i] is None
+            if metric_names is None:
+                assert df.get("values")[0][i] is None
+                assert df.get("values")[1][i] is None
+            else:
+                assert df.get("values")[metric_names[0]][i] is None
+                assert df.get("values")[metric_names[1]][i] is None
     else:
         for i in range(2):
-            assert df.values_0[i] is None
-            assert df.values_1[i] is None
+            if metric_names is None:
+                assert df.values_0[i] is None
+                assert df.values_1[i] is None
+            else:
+                assert df.get(f"values_{metric_names[0]}")[i] is None
+                assert df.get(f"values_{metric_names[1]}")[i] is None
```

Comment:
I am concerned that this test case is becoming more complex. Do you think we should split them into new test cases?